kratos/solving_strategies/schemes/scheme.h

```diff
@@ -737,6 +761,17 @@ class Scheme
      * @param EquationId The ID's of the element degrees of freedom
      * @param rCurrentProcessInfo The current process info instance
      */
+    virtual void CalculateSystemContributions(
+        Element::Pointer pCurrentElement,
```

Comment:
is there a reason we pass a pointer and not a reference?
To my understanding the Scheme should not modify the pointer right?